### PR TITLE
Task377 questions

### PR DESCRIPTION
**Pull request by nibivi77**
# Description 📝

We created a Redux store for the questions. On the home page, in the Redux dev tool, we can see all questions are fetched from the Redux store.

Unit tests were also added for both getQuestions and getQuestion.

Branch name: Task377Questions

(Please give feedback on this. Too big of a PR? Not coded properly? Not efficient? Anything helps me out here.)

## Taiga Link 🔗

https://taiga.dts-stn.com/project/ericdube-hfp-shared-backlog/task/377?kanban-status=130

## Checklist ✅
- [ ] created story for all visual components
- [ x] created or modified necessary unit tests
- [ ] created or modified e2e tests
- [ ] modified README or any applicable documentation on the changes you made

### CODE REVIEW

File: src/redux/reducers/questions/questionsDataReducer.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        }
+        if (Array.isArray(data) && data.length > 0) {
+          let newQuestionsMap = { ...state.questionsMap };
+          data.forEach((value) => {
+            if (!newQuestionsMap[value.id]) {
+              newQuestionsMap[value.id] = {
+                isEligible: true,
+                isSelected: false,
+              };
+            }
```
Comment (var-kyle) [inline src/redux/reducers/questions/questionsDataReducer.js]: Maybe I don't get this, but questions shouldn't be eligible or selected, should they?

Comment (nibivi77) [reply src/redux/reducers/questions/questionsDataReducer.js]: You are probably right. I was just unsure how we wanted to handle the questions, so I left it like the benefits. I will make the change.

File: src/redux/reducers/questions/questionsDataReducer.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          });
+
+          return {
+            ...state,
+            isFetching: false,
+            fetchFailed: false,
+            fetchFailedReason: "",
+            fetchFailedObj: {},
+            questionsMapFr: newQuestionsMapFr,
+          };
```
Comment (var-kyle) [inline src/redux/reducers/questions/questionsDataReducer.js:90]: If the question map is part of the state, is it necessary to return it separately as well?

File: src/redux/reducers/questions/questionsDataReducer.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          });
+
+          return {
+            ...state,
+            isFetching: false,
+            fetchFailed: false,
+            fetchFailedReason: "",
+            fetchFailedObj: {},
+            questionsMap: newQuestionsMap,
+          };
```
Comment (var-kyle) [inline src/redux/reducers/questions/questionsDataReducer.js:62]: If the question map is part of the state, is it necessary to return it separately as well?

**Review by Moro-Code — APPROVED** [review]

**Pull request merged**